src/Dependencies/Collections/ImmutableSegmentedDictionary`2+PrivateInterlocked.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.Collections
+{
+    internal readonly partial struct ImmutableSegmentedDictionary<TKey, TValue>
+    {
+        internal static class PrivateInterlocked
```

Comment:
>internal [](start = 8, length = 8)

`private`?

Reply:
Can't; it's used by `RoslynImmutableInterlocked`.

Reply:
I don't see point of this type. I'd rather have all members of this type and members of RoslynImmutableInterlocked moved to `ImmutableSegmentedDictionary` static type.

---
In reply to: [561175363](https://github.com/dotnet/roslyn/pull/50189#discussion_r561175363) [](ancestors = 561175363)

Reply:
`ImmutableSegmentedDictionary` doesn't have access to the `_dictionary` field.